roseau/load_flow/_solvers.py

```diff
@@ -162,9 +163,24 @@ def save_matrix(self, prefix: str) -> None:
         """
         raise NotImplementedError(f"save_matrix() is not implemented for solver {self.name!r}.")
 
-    def current_jacobian(self) -> FloatMatrix:
+    def jacobian(self) -> FloatMatrix:
         """Get the jacobian of the current iteration (useful for debugging)."""
-        raise NotImplementedError(f"current_jacobian() is not implemented for solver {self.name!r}.")
+        raise NotImplementedError(f"jacobian() is not implemented for solver {self.name!r}.")
+
+    @property
+    def variables(self) -> FloatArray:
+        """Get the variables of the current iteration (useful for debugging)."""
+        return self._cy_solver.get_variables()
+
+    @variables.setter
+    def variables(self, variables: FloatArrayLike1D) -> None:
+        """Set the independent variables (useful for debugging)."""
+        self._cy_solver.set_variables(np.array(variables, dtype=np.float64))
+
+    @property
+    def residuals(self) -> FloatArray:
+        """Get the residuals of the current iteration (useful for debugging)."""
+        raise NotImplementedError(f"residuals() is not implemented for solver {self.name!r}.")
```

Comment:
One more minor thing, since this is a property
```suggestion
        raise NotImplementedError(f"residuals is not implemented for solver {self.name!r}.")
```